### PR TITLE
Test public / private flags can be set declaratively

### PR DESCRIPTION
**Pull request by ingenieroariel**

### CODE REVIEW

File: all-tests.nix [hunk 1]
```diff
@@ -100,8 +100,20 @@ in
           client.succeed("ob thunk pack ~/code/myapp")
           client.succeed("grep -qF 'git' ~/code/myapp/default.nix")
           client.succeed("grep -qF 'myorg' ~/code/myapp/git.json")
+          client.succeed("ob thunk unpack ~/code/myapp")
```
Comment (3noch) [inline all-tests.nix]: Let's slap `-v` on these :smile:

File: all-tests.nix [hunk 2]
```diff
@@ -100,8 +100,20 @@ in
...
       
+      with subtest("test obelisk can set the public / private flag"):
+          client.succeed("ob thunk pack ~/code/myapp --private")
+          client.fail("""grep -qF '"private": practice' ~/code/myapp/git.json""")
+          client.succeed("""grep -qF '"private": true' ~/code/myapp/git.json""")
+          client.succeed("ob thunk unpack ~/code/myapp")
+          client.succeed("ob thunk pack ~/code/myapp --public")
+          client.succeed("""grep -qF '"private": false' ~/code/myapp/git.json""")
+          client.succeed("ob thunk unpack ~/code/myapp")
```
Comment (3noch) [inline all-tests.nix]: We also need to test that doing something with nix on the thunk works. Like `nix eval "import  ~/code/myapp"` or something (check that this actually forces the nix to pull down the thunk).

Comment (ingenieroariel) [reply all-tests.nix]: done in https://github.com/obsidiansystems/obelisk/pull/619/commits/2e0b7698b36c1301ac73113c7c2cda6460cea3e5

Comment (3noch) [reply all-tests.nix]: I think you meant the `-v` change was done not this one?

Comment (ingenieroariel) [reply all-tests.nix]: right

Comment (ingenieroariel) [reply all-tests.nix]: something with the thunk done in  https://github.com/obsidiansystems/obelisk/pull/619/commits/1f3ee0ee6ce9d572a4a7de8e8658bd4e7a3af343

File: all-tests.nix [hunk 3]
```diff
@@ -46,6 +46,12 @@ in
     testScript =
       let
         privateKeyFile = pkgs.writeText "id_rsa" ''${snakeOilPrivateKey}'';
+        thunkableSample = pkgs.writeText "default.nix" ''
+        let pkgs = import <nixpkgs> {}; in pkgs.git
+        '';
+        invalidThunkableSample = pkgs.writeText "default.nix" ''
+        let pkgs = import <nixpkgs> {}; in pkgtypo.git
```
Comment (3noch) [inline all-tests.nix]: Indent these multiline strings?

File: all-tests.nix [hunk 4]
```diff
@@ -106,11 +115,28 @@ in
...
+          client.succeed(
+              "cp ${invalidThunkableSample} ~/code/myapp/default.nix"
+          )
+          client.succeed("cd ~/code/myapp && git add .")
+          client.succeed('git config --global user.email "you@example.com"')
+          client.succeed('git config --global user.name "Your Name"')
+          client.succeed('cd ~/code/myapp && git commit -m "Bad commit"')
+          client.succeed("cd ~/code/myapp && git push -u origin bad")
+          client.succeed("ob -v thunk pack ~/code/myapp --public")
+          client.fail("nix-build  ~/code/myapp")
```
Comment (3noch) [inline all-tests.nix]: Why should this fail?

Comment (3noch) [reply all-tests.nix]: Oh now I see. Hehe

**Pull request merged**